### DISCUSSION TITLE
maddy 0.5.0

### DISCUSSION THREAD

**Discussion by foxcpp** [emersion/maddy#381] [General]
## New features

* Experimental: Built-in ACME client (GH #3)

Currently supports only dns-01 challenge with a limited set of
providers. See documentation for details.

* S3-backed storage for message contents (GH #304)

* Local sender authorization (GH #268)

* LDAP BindDN authentication (GH #273)

* storage/imapsql: Implement auth_map
* storage/imapsql: Implement delivery_map

This functionality allows imapsql storage backend to be correctly
used with non-email-based authentication providers.
In particular, this unbreaks PAM and shadow modules.

* Implement table.chain module

* Implement table.email_localpart as a helper to strip domain from emails

## Improvements

* Implement client timeouts for target.remote and target.smtp
* endpoint/smtp: Add max_header_size

## Fixes

* check/spf: Change default action for softfail to 'ignore'
* endpoint/smtp: Allow to change the line length limit enforced by go-smtp
* table/sql_query: Allow to use numbered parameters in queries
* auth/plain_separate: Make configuration directives actually work
* table/file: Allow table to be created without specifying files in inline args
* config/tls: Fix custom loader configuration reading

## Removed functionality

* check/dns: Mark require_matching_echo as deprecated
* config/tls: Remove deprecated "tls CERT KEY" syntax
* Remove deprecated 0.3 module name aliases

<hr /><em>This discussion was created from the release <a href='https://github.com/foxcpp/maddy/releases/tag/v0.5.0'>maddy 0.5.0</a>.</em>

**Comment by rajveermalviya** [comment]
unable to find the docker image.
are they manually pushed?

https://hub.docker.com/r/foxcpp/maddy/tags?page=1&ordering=last_updated

**Reply by rajveermalviya** [reply]
cc @foxcpp

**Reply by foxcpp** [reply]
Woops. Yes, they are and I forgot about it. Give me 10 minutes.

**Reply by foxcpp** [reply]
There you go.
https://hub.docker.com/layers/foxcpp/maddy/0.5.0/images/sha256-1d0b5e784485377c0730feb55b0ec517068d7e887440bd6b377f18760ab7620f?context=explore

**Comment by theduke** [comment]
Thanks for the S3 backend!

Can I migrate by just copying the files into the target bucket?

**Reply by foxcpp** [reply]
You should be able to. I haven't tested such migration though.
